### PR TITLE
YR/-Convert-IN116-into-BC111/CIAC-10243

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-10243

## Description
Convert the IN116 validation, to the new format. Now it will be called BC111

### CODE REVIEW

File: demisto_sdk/commands/validate/validators/BC_validators/BC111_new_required_argument_script.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+ContentTypes = Script
+
+
+class NewRequiredArgumentScriptValidator(BaseValidator[ContentTypes]):
+    error_code = "BC111"
+    description = (
+        "Validate that no new *required* argument are added to an existing script."
+    )
+    rationale = "Adding a new argument to an existing script and defining it as *required* or changing an non-required argument to be required will break backward compatibility."
```
Comment (JasBeilin) [inline demisto_sdk/commands/validate/validators/BC_validators/BC111_new_required_argument_script.py:20]: Lets make it cleaner and clearer with mentioning of default value.

File: demisto_sdk/commands/validate/validators/BC_validators/BC111_new_required_argument_script.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                            for arg in old_content_item.args  # type: ignore
+                            if arg.name == current_arg_name
+                        ),
+                        None,
+                    )
+
+                    if (
+                        # If the argument is new or the argument was not required before
+                        not old_corresponding_arg
+                    ) or not old_corresponding_arg.required:
```
Comment (JasBeilin) [inline demisto_sdk/commands/validate/validators/BC_validators/BC111_new_required_argument_script.py:48]: can we simplify this?

**Review by JasBeilin — CHANGES_REQUESTED** [review]
Great job, see my comments as we discussed.

File: demisto_sdk/commands/validate/tests/BC_validators_test.py [hunk 3]
```diff
@@ -946,57 +946,82 @@ def test_IsValidToversionOnModifiedValidator_is_valid(content_items, old_content
...
 
 
-def test_HaveCommandsOrArgsNameChangedValidator__script__fails():
+@pytest.mark.parametrize(
+    "new_args, breaking_arg",
+    [
+        pytest.param(
+            [{"name": "arg1", "required": False}, {"name": "arg2", "required": True}],
+            "arg2",
+            id="two args, new one required",
```
Comment (JasBeilin) [inline demisto_sdk/commands/validate/tests/BC_validators_test.py:955]:
```suggestion
            id="new required param",
```

File: demisto_sdk/commands/validate/tests/BC_validators_test.py [hunk 4]
```diff
@@ -946,57 +946,82 @@ def test_IsValidToversionOnModifiedValidator_is_valid(content_items, old_content
...
+        ),
+        pytest.param(
+            [
+                {"name": "arg1", "required": True},
+            ],
+            "arg1",
+            id="one arg, required",
```
Comment (JasBeilin) [inline demisto_sdk/commands/validate/tests/BC_validators_test.py:962]:
```suggestion
            id="changed required",
```

File: demisto_sdk/commands/validate/tests/BC_validators_test.py [hunk 5]
```diff
@@ -946,57 +946,82 @@ def test_IsValidToversionOnModifiedValidator_is_valid(content_items, old_content
...
-def test_HaveCommandsOrArgsNameChangedValidator__script__passes():
+@pytest.mark.parametrize(
+    "new_args",
+    [
+        pytest.param(
+            [
+                {"name": "arg1", "required": False},
+                {"name": "arg2", "required": False},
+            ],
+            id="two args, both not required",
```
Comment (JasBeilin) [inline demisto_sdk/commands/validate/tests/BC_validators_test.py:997]:
```suggestion
            id="non required args",
```

File: demisto_sdk/commands/validate/tests/BC_validators_test.py [hunk 6]
```diff
@@ -946,57 +946,82 @@ def test_IsValidToversionOnModifiedValidator_is_valid(content_items, old_content
...
+        - An old script with 2 args, one required and one not.
+        - A new script:
+        case 1: with 2 args, both not required.
+        case 2: with 2 args, both required, and the second one has a default value.
     When
         calling the NewRequiredArgumentScriptValidator.
     Then
-        - Make sure the validation passes, since the new required arguments have a default value, or are not new.
+        - Make sure the validation passes:
+        case 1: the first arg was changed to be not required which is allowed.
```
Comment (JasBeilin) [inline demisto_sdk/commands/validate/tests/BC_validators_test.py:1019]: Is it what we are checking here? both args are not required

**Review by JasBeilin — COMMENTED** [review]
Looks great, I think we should improve our test's descriptions - see my comments.

File: demisto_sdk/commands/validate/tests/BC_validators_test.py (hunk 3 above)
Comment (RosenbergYehuda) [reply demisto_sdk/commands/validate/tests/BC_validators_test.py:955]: fixed

File: demisto_sdk/commands/validate/tests/BC_validators_test.py (hunk 5 above)
Comment (RosenbergYehuda) [reply demisto_sdk/commands/validate/tests/BC_validators_test.py:997]: fixed

File: demisto_sdk/commands/validate/tests/BC_validators_test.py (hunk 4 above)
Comment (RosenbergYehuda) [reply demisto_sdk/commands/validate/tests/BC_validators_test.py:962]: fixed

File: demisto_sdk/commands/validate/tests/BC_validators_test.py [hunk 7]
```diff
@@ -946,57 +946,82 @@ def test_IsValidToversionOnModifiedValidator_is_valid(content_items, old_content
...
+                {"name": "arg2", "required": False},
+            ],
+            id="two args, both not required",
+        ),
+        pytest.param(
+            [
+                {"name": "arg1", "required": True},
+                {"name": "arg2", "required": True, "defaultvalue": "test"},
+            ],
+            id="two args, both required, second one with default value",
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply demisto_sdk/commands/validate/tests/BC_validators_test.py:1004]: fixed

File: demisto_sdk/commands/validate/tests/BC_validators_test.py (hunk 6 above)
Comment (RosenbergYehuda) [reply demisto_sdk/commands/validate/tests/BC_validators_test.py:1019]: Yes. validationg that that is OK.

**Review by JasBeilin — APPROVED** [review]
Nice!

**Pull request merged**